src/modules/keycloak-administration/domain/keycloak-user.service.ts

```diff
@@ -163,6 +163,56 @@ export class KeycloakUserService {
         }
     }
 
+    public async updateUser(
+        username: string,
+        oxUserName: OXUserName,
+        oxContextName: OXContextName,
+    ): Promise<Result<void, DomainError>> {
+        // Get authed client
+        const kcAdminClientResult: Result<KeycloakAdminClient, DomainError> =
+            await this.kcAdminService.getAuthedKcAdminClient();
+
+        if (!kcAdminClientResult.ok) {
+            return kcAdminClientResult;
+        }
+
+        // Check for existing user
+        const filter: FindUserFilter = {
+            username: username,
+        };
+
+        const findResult: Result<User<true>, DomainError> = await this.findOne(filter);
+
+        if (!findResult.ok) {
+            return {
+                ok: false,
+                error: findResult.error,
+            };
+        }
+
+        const foundUser: User<true> = findResult.value;
+        const newExternalSystemIDs: ExternalSystemIDs = {
+            ID_ITSLEARNING: foundUser.externalSystemIDs.ID_ITSLEARNING,
+            ID_OX: oxUserName + '@' + oxContextName,
+        };
+
+        const userRepresentation: UserRepresentation = {
+            //only attributes shall be updated here for this event
+            attributes: newExternalSystemIDs,
+        };
```

Comment:
Be very careful here. Attributes might contain additional data, which you will just throw away here. Team 2 also uses the attributes to store the reason if a user is deactivated.

Reply:
Adjustment should fix this, the ITS_LEARNING key stays as is when attribute for OX is updated.